### PR TITLE
Flush 009Q.u on 009Q.z for correct url and provides data

### LINKED ISSUE #270: Homepage URL not transformed correctly for some entries
See e.g. http://lobid.org/organisations/DE-38M?format=json.:

```json
 "url" : "https://www.livivo.de/?idb=ZBMED",
  "provides" : "https://www.livivo.de/?idb=ZBMED"
```

[XML source](http://services.dnb.de/sru/bib?operation=searchRetrieve&query=isl%3DDE-38M&recordSchema=PicaPlus-xml&version=1.1) looks correct. The homepage URL is found in 009Q, with code "A" in subfield z. (See also the [documentation of this field](http://sigel.staatsbibliothek-berlin.de/de/vergabe/adressenformat/856/).

```xml
<ppxml:tag id="009Q" occ="">
  <ppxml:subf id="u">http://www.zbmed.de</ppxml:subf>
  <ppxml:subf id="z">A</ppxml:subf>
</ppxml:tag>
<ppxml:tag id="009Q" occ="">
  <ppxml:subf id="u">http://www.livivo.de/?idb=ZBMED</ppxml:subf>
  <ppxml:subf id="z">B</ppxml:subf>
</ppxml:tag>
<ppxml:tag id="009Q" occ="">
  <ppxml:subf id="u">https://www.livivo.de/?idb=ZBMED</ppxml:subf>
  <ppxml:subf id="x">Weiterer Katalogzugang</ppxml:subf>
  <ppxml:subf id="z">W</ppxml:subf>
</ppxml:tag>
```

Expected outcome of the transformation:

 ```json
 "url" : "http://www.zbmed.de",
  "provides" : "https://www.livivo.de/?idb=ZBMED"
```

### PR DESCRIPTION
**Pull request by fsteeg**
Avoids picking the wrong URL for the 009Q.z type marker

Will resolve #270

### CODE REVIEW

File: app/transformation/morph-enriched.xml [hunk 1]
```diff
@@ -17,8 +17,13 @@
 						<regexp match="(www(.*))" format="http://${1}"/>
 						<replace pattern="\s+" with=""/>
 					</data>
-					<combine value="${hp}" name="$[group_name]" sameEntity="true">
-						<choose>
+					<combine value="${hp}" name="$[group_name]" sameEntity="true" flushWith="009Q.z" reset="true">
+						<if>
+							<data source="009Q.z" name="">
```
Comment (dr0i) [inline app/transformation/morph-enriched.xml]: I think you don't need to set "name". Just leave it out.

Comment (fsteeg) [inline app/transformation/morph-enriched.xml]: Yeah, I thought so too, but it's required in the schema.

Comment (dr0i) [inline app/transformation/morph-enriched.xml]: No, it's optional, see https://github.com/hbz/metafacture-core/blob/master/src/main/resources/schemata/metamorph.xsd#L472

Comment (fsteeg) [inline app/transformation/morph-enriched.xml]: Oh, the `name` on the `data` element, right! Sorry, thought you meant the `name` on `combine`.

**Comment by dr0i** [comment]
+1

**Pull request merged**